joy2twist/src/joy2twist_node.cpp

```diff
@@ -66,16 +66,81 @@ void Joy2TwistNode::load_parameters()
-  this->get_parameter<int>("button_index_map.axis.linear_y", button_index_.linear_y);
-  this->get_parameter<int>("button_index_map.dead_man_switch", button_index_.dead_man_switch);
-  this->get_parameter<int>("button_index_map.fast_mode", button_index_.fast_mode);
-  this->get_parameter<int>("button_index_map.slow_mode", button_index_.slow_mode);
-  this->get_parameter<int>("button_index_map.e_stop_reset", button_index_.e_stop_reset);
-  this->get_parameter<int>("button_index_map.e_stop_trigger", button_index_.e_stop_trigger);
-  this->get_parameter<int>(
-    "button_index_map.enable_e_stop_reset", button_index_.enable_e_stop_reset);
+  RawInputIndex raw_input_index{};
+
+  this->get_parameter<std::string>("input_index_map.axis.angular_z", raw_input_index.angular_z);
+  this->get_parameter<std::string>("input_index_map.axis.linear_x", raw_input_index.linear_x);
+  this->get_parameter<std::string>("input_index_map.axis.linear_y", raw_input_index.linear_y);
+  this->get_parameter<std::string>(
+    "input_index_map.dead_man_switch", raw_input_index.dead_man_switch);
+  this->get_parameter<std::string>("input_index_map.fast_mode", raw_input_index.fast_mode);
+  this->get_parameter<std::string>("input_index_map.slow_mode", raw_input_index.slow_mode);
+  this->get_parameter<std::string>("input_index_map.e_stop_reset", raw_input_index.e_stop_reset);
+  this->get_parameter<std::string>(
+    "input_index_map.e_stop_trigger", raw_input_index.e_stop_trigger);
+  this->get_parameter<std::string>(
+    "input_index_map.enable_e_stop_reset", raw_input_index.enable_e_stop_reset);
+
+  parse_joy_inputs(raw_input_index);
+}
+
+void Joy2TwistNode::parse_joy_inputs(const RawInputIndex & raw_input_index)
+{
+  input_index_.angular_z = JoyInput::from_string(raw_input_index.angular_z);
+  input_index_.linear_x = JoyInput::from_string(raw_input_index.linear_x);
+  input_index_.linear_y = JoyInput::from_string(raw_input_index.linear_y);
+  input_index_.dead_man_switch = JoyInput::from_string(raw_input_index.dead_man_switch);
+  input_index_.fast_mode = JoyInput::from_string(raw_input_index.fast_mode);
+  input_index_.slow_mode = JoyInput::from_string(raw_input_index.slow_mode);
+  input_index_.e_stop_reset = JoyInput::from_string(raw_input_index.e_stop_reset);
+  input_index_.e_stop_trigger = JoyInput::from_string(raw_input_index.e_stop_trigger);
+  input_index_.enable_e_stop_reset = JoyInput::from_string(raw_input_index.enable_e_stop_reset);
+}
+
+float Joy2TwistNode::get_joy_input(
+  const MsgJoy::SharedPtr joy_msg, const JoyInput & joy_input) const
+{
+  float value = 0.0f;
+
+  if (joy_input.type == JoyInput::Type::AXIS) {
+    value = joy_msg->axes.at(joy_input.index);
+  } else if (joy_input.type == JoyInput::Type::BUTTON) {
+    value = static_cast<float>(joy_msg->buttons.at(joy_input.index));
+  } else {
+    throw std::invalid_argument("Invalid JoyInput type");
+  }
+
+  if (joy_input.is_inverted) {
+    value *= -1;
+  }
+
+  return value;
```

Comment:
```suggestion
  float negation_factor = joy_input.is_inverted ? -1.0f : 1.0f;

  if (joy_input.type == JoyInput::Type::AXIS) {
    return joy_msg->axes.at(joy_input.index) * negation_factor;
  }
  
  if (joy_input.type == JoyInput::Type::BUTTON) {
    return static_cast<float>(joy_msg->buttons.at(joy_input.index)) * negation_factor;
  }
  
  throw std::invalid_argument("Invalid JoyInput type");
```
Personally would do it this way to reduce if-else statements, but it's up to you

Reply:
Yep, good idea as this reduces code cognitive complexity